crates/starknet-devnet/tests/test_v3_transactions.rs

```diff
@@ -0,0 +1,472 @@
+                    match declaration.send().await.unwrap_err() {
+                        starknet_rs_accounts::AccountError::Provider(
+                            ProviderError::StarknetError(StarknetError::InsufficientMaxFee),
+                        ) => {}
+                        starknet_rs_accounts::AccountError::Provider(ProviderError::Other(
+                            other,
+                        )) if other.to_string().contains("is lower than the actual gas price") => {}
+                        other => panic!("Unexpected error: {:?}", other),
+                    }
+                }
+                Action::AccountDeployment(salt) => {
+                    let mut account_deployment =
+                        AccountDeploymentV3::new(*salt, account_factory.unwrap());
+                    if let Some(gas_units) = gas_units {
+                        account_deployment = account_deployment.gas(gas_units);
+                    }
+                    if let Some(gas_price) = gas_price {
+                        account_deployment = account_deployment.gas_price(gas_price);
+                    }
+                    match account_deployment.send().await.unwrap_err() {
+                        starknet_rs_accounts::AccountFactoryError::Provider(
+                            ProviderError::StarknetError(StarknetError::InsufficientMaxFee),
+                        ) => {}
+                        starknet_rs_accounts::AccountFactoryError::Provider(
+                            ProviderError::Other(other),
+                        ) if other.to_string().contains("is lower than the actual gas price") => {}
+                        other => panic!("Unexpected error: {:?}", other),
+                    }
+                }
+                Action::Execution(calls) => {
+                    let mut execution = ExecutionV3::new(calls.clone(), account.unwrap());
+                    if let Some(gas_units) = gas_units {
+                        execution = execution.gas(gas_units);
+                    }
+                    if let Some(gas_price) = gas_price {
+                        execution = execution.gas_price(gas_price);
+                    }
+                    let transaction_result = execution.send().await;
+
+                    match transaction_result {
+                        Ok(InvokeTransactionResult { transaction_hash }) => {
+                            let receipt = account
+                                .unwrap()
+                                .provider()
+                                .get_transaction_receipt(transaction_hash)
+                                .await
+                                .unwrap();
+                            let execution_result = receipt.receipt.execution_result();
+                            match execution_result {
+                                ExecutionResult::Reverted { reason } => {
+                                    assert!(reason.contains("Insufficient max L1 gas"));
```

Comment:
Why not use `server::test_utils::assert_contains`?

Reply:
[7e4a1bb](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/625/commits/7e4a1bb1169a9eeb20740e60edb18a15a1bfe4a7)